src/main/scala/cosas/fns/functions.scala

```diff
@@ -117,8 +99,21 @@ trait AnyAt2 extends Any with AnyAt {
   def apply(in1: X1, in2: X2): Y
 }
 
+trait AnyAt3 extends Any with AnyAt {
+
+  type DepFn <: AnyDepFn3
+  type X1 <: DepFn#In1; type X2 <: DepFn#In2; type X3 <: DepFn#In3
+
+  def apply(in1: X1, in2: X2, in3: X3): Y
+}
```

Comment:
So in the end what's the point of these traits? I think that having only `AppX` classes is just enough.